### DISCUSSION TITLE
API check returns the whole string every time

### DISCUSSION THREAD

**Discussion by JimmyInsane** [LostRuins/koboldcpp#932] [Q&A]
I'm currently writing a small python script to access the API.
Currently on one thread I start the generation while the other periodically checks if there is new content.
I'm using a post on this localhost:5001/api/extra/generate/check
But this always returns the whole current string and not just the part that hasn't been polled yet.

For example the full output string would be "Hey there, I am your personal AI"
The result would be:
"Hey there,"
"Hey there, I am your"
"Hey there, I am your personal AI"

Which I guess would be fine if i just update a div in a webpage but for a console application i have to carriage return every time I get a message.

Is this intended?

**Comment by LostRuins** [comment]
Yes, this is intended. The polling is intended to check progress on an ongoing sync generate.

For SSE streaming, you can use the /api/extra/generate/stream instead.
